core/src/smartcontracts/isi/permissions/mod.rs

```diff
@@ -513,16 +513,23 @@ mod tests {
         let instruction: Instruction = Pair::new(
             TransferBox::new(
                 asset_id("btc", "crypto", "seller", "company"),
-                Expression::Add(Add::new(
-                    Expression::Query(
-                        FindAssetQuantityById::new(AssetId::new(
-                            AssetDefinitionId::from_str("btc2eth_rate#exchange").expect("Valid"),
-                            AccountId::from_str("dex@exchange").expect("Valid"),
-                        ))
-                        .into(),
-                    ),
-                    10_u32,
-                )),
+                EvaluatesTo::new_evaluates_to_value(
```

Comment:
Reminds me of the tower of doom.

Reply:
I'm not in the context of this game)

But I agree, this looks not so good, but I can't find a better name

Reply:
Also known as [pyramid of doom](https://en.wikipedia.org/wiki/Pyramid_of_doom_(programming)).

Reply:
Does SDK developers need to wrap there queries into all this `EvaluatesTo` now or it only for internal use?

Reply:
Only internal usage. We got input from SDK using some (de-)serialization. Stong-typing affects only internal usage

Reply:
> Also known as [pyramid of doom](https://en.wikipedia.org/wiki/Pyramid_of_doom_(programming)).
Oh, now I got it